nautobot/extras/tests/test_context_managers.py

```diff
@@ -194,3 +197,97 @@ def test_change_log_thread_safe(self):
+                location = Location(name="Test Location 1", location_type=location_type, status=location_status)
+                location.save()
+
+        location = Location.objects.get(name="Test Location 1")
+        oc_list = get_changes_for_model(location).order_by("pk")
+        self.assertEqual(len(oc_list), 1)
+        self.assertEqual(oc_list[0].changed_object, location)
+        self.assertEqual(oc_list[0].action, ObjectChangeActionChoices.ACTION_CREATE)
+
+    def test_delete(self):
+        """Test that deletes raise an exception"""
+        location_type = LocationType.objects.get(name="Campus")
+        location_status = Status.objects.get_for_model(Location).first()
+        with self.assertRaises(ValueError):
+            with web_request_context(self.user):
+                with deferred_change_logging_for_bulk_operation():
+                    location = Location(name="Test Location 1", location_type=location_type, status=location_status)
+                    location.save()
+                    location.delete()
+
+    def test_create_then_update(self):
+        """Test that a create followed by an update is logged as a single create"""
+        location_type = LocationType.objects.get(name="Campus")
+        location_status = Status.objects.get_for_model(Location).first()
+        with web_request_context(self.user):
+            with deferred_change_logging_for_bulk_operation():
+                location = Location(name="Test Location 1", location_type=location_type, status=location_status)
+                location.save()
+                location.description = "changed"
+                location.save()
+
+        oc_list = get_changes_for_model(location)
+        self.assertEqual(len(oc_list), 1)
+        self.assertEqual(oc_list[0].action, ObjectChangeActionChoices.ACTION_CREATE)
+        snapshots = oc_list[0].get_snapshots()
+        self.assertIsNone(snapshots["prechange"])
+        self.assertIsNotNone(snapshots["postchange"])
+        self.assertIsNone(snapshots["differences"]["removed"])
+        self.assertEqual(snapshots["differences"]["added"]["description"], "changed")
+
+    def test_bulk_edit(self):
+        """Test that edits to multiple objects are correctly logged"""
+        location_type = LocationType.objects.get(name="Campus")
+        location_status = Status.objects.get_for_model(Location).first()
+        locations = [
+            Location(name=f"Test Location {i}", location_type=location_type, status=location_status)
+            for i in range(1, 4)
+        ]
+        Location.objects.bulk_create(locations)
+        with web_request_context(self.user):
+            with deferred_change_logging_for_bulk_operation():
+                for location in locations:
+                    location.description = "changed"
+                    location.save()
+
+        oc_list = get_changes_for_model(Location)
+        self.assertEqual(len(oc_list), 3)
```

Comment:
I suspect this is failing in CI because we have an `ObjectChangeFactory` now so there may be pre-existing changes for the Location model. Probably need to filter `oc_list` more explicitly to only include the given `locations` instances?